Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
+    EyelinkDrawClearScreen(eyewin, el); 
     clearScreen=0;
 end
-% New video data from eyelink?
-if newcamimage
-    % Video callback from Eyelink: We have a 'eyewidth' by 'eyeheight' pixels
-    % live eye image from the Eyelink system. Each pixel is encoded as a 4 byte
-    % RGBA pixel with alpha channel set to a constant value of 255 and the RGB
-    % channels encoding a 1-Byte per channel R, G or B color value. The
-    % given 'eyeimgptr' is a specially encoded memory pointer to the memory
-    % buffer inside Eyelink() that encodes the image.
+
+if newcamimage      % New image frame received from EyeLink camera stream
+    % Image has dimensions: 'eyewidth' by 'eyeheight' in pixel units
+    % Each pixel is encoded as a 4 byte RGBA pixel with A=255 always
+    % RGB channels each encode a 1-Byte per channel R, G or B color value.
+    % 'eyeimgptr' is a memory pointer to the buffer inside Eyelink() that
+    % encodes the image.
     eyeimgptr = callArgs(2);
     eyewidth  = callArgs(3);
     eyeheight = callArgs(4);
     
-    % Create a new PTB texture of proper format and size and inject the 4
-    % channel RGBA color image from the Eyelink memory buffer into the texture.
-    % Return a standard PTB texture handle to it. If such a texture already
-    % exists from a previous invocation of this routiene, just recycle it for
-    % slightly higher efficiency:
+    % Creates a new or reuses an existing PTB texture for the cam image
     eyelinktex = Screen('SetOpenGLTextureFromMemPointer', eyewin, eyelinktex, eyeimgptr, eyewidth, eyeheight, 4, 0, [], GL_RGBA8, GL_RGBA, hostDataFormat);
 end
 
-% If we're in imagemodedisplay, draw eye camera image texture centered in
-% window, if any such texture exists, also draw title if it exists.
-if ~isempty(eyelinktex) && ineyeimagemodedisplay==1
-    imgtitle=EyelinkDrawCameraImage(eyewin, el, eyelinktex, imgtitle,newImage);
+if ~isempty(eyelinktex) && ineyeimagemodedisplay==1     % Draw cam image and caption
+    imgtitle=EyelinkDrawCameraImage(eyewin, el, eyelinktex, imgtitle, newImage);
 end
 
-% Draw calibration target, if any is specified:
-if ~isempty(calxy)
+if ~isempty(calxy)  % Draw Cal Target
     drawInstructions=0;
     EyelinkDrawCalibrationTarget(eyewin, el, calxy);
+    if strcmpi(el.calTargetType, 'video')
+        return;
+    end
 end
 
-% Need to draw instructions?
-if drawInstructions==1
-    EyelinkDrawInstructions(eyewin, el,msg);
-    drawInstructions=0;
+if drawInstructions == 1  % Draw Instructions
+    EyelinkDrawInstructions(eyewin, el, msg);
+    drawInstructions = 0;
 end
 
-% Show it: We disable synchronization of Matlab to the vertical retrace.
-% This way, display update itself is still synced and tear-free, but we
-% don't waste time waiting for swap completion. Potentially higher
-% performance for calibration displays and eye camera image updates...
-% Neither do we erase buffer
-Screen('Flip', eyewin, [], 1, 1);
+dontsync = 1;
+dontclear = 0;
```

Comment:
Why was dontclear changed from 1 to 0? No longer needed due to your other code changes that cause always a redraw of everything? Or some other reason?

Reply:
in current state, dontclear=0 wipes instructions when showing camera image stream (expected behaviour). added comment that dontclear=1 allows instructions to persist during camera image stream. resolved in 9e17d4f50104aab97855289e617b8627d7442617